### DISCUSSION TITLE
Question about searching nodes near a location

### DISCUSSION THREAD

**Discussion by ChanWhanPark** [valhalla/valhalla#4284] [Q&A]
I would like to ask about the algorithm for navigating Node on Valhalla engine.

When searching for a path, if there is no node nearby, it is searched to that point or Vertex, but if the node is located within a certain range (about between 5 ~ 10 meters), it seems that the path is searched to that node. Here's an example.

+ In the current example image, the red dot is not a vertex belonging to the OSM way. As a parking lot area Entry Point, if within the parking area, request route guidance to Entry Point and create additional geometry to the destination.

**The right case**
![image](https://github.com/valhalla/valhalla/assets/68843687/533cac24-8b84-4dcb-be7b-0c50f7a6c40a)
This is the route to the parking lot area after guiding to that point.

**The Wrong case**
![image](https://github.com/valhalla/valhalla/assets/68843687/72dc69d5-a0c8-4a30-ae66-7712c72b498f)
It has navigated the path to Node instead of that point. It is about 5 meters from that point to Node.

Even if setting the red dot to the middle Vertex of the OSM way, the result is the same.

I guess I can modify the loki area of the loki library or config.json of bench/meili, but it seems that I need a guide on which parameters to touch. I would like to ask if there are any parameters that can narrow the search scope while specifying the location or if there is a way to check it while modifying it in the loki library.

Thank you always for your guidance.

**Comment by nilsnolde** [comment]
Yes you’re right, we do snap to graph nodes and the default is indeed 5 m. It’s a pretty sensible default, but if you’re in accurate control of the input locations you can of course lower it. Look at a location‘s „node_snap_tolerance“ parameter: http://valhalla.github.io/valhalla/api/turn-by-turn/api-reference/#locations. 

I’d generally recommend you have a more detailed read on the API docs to see what’s possible.
